app/models/concerns/orchestration/dhcp.rb

```diff
@@ -92,14 +92,7 @@ def boot_server
       # we no longer convert the boot server to IPv4 - smart proxy modules are required to do so if they don't support hostname
       bs
     else
-      begin
-        Foreman::Deprecation.deprecation_warning('1.25', "DHCP proxy does not report dhcp_filename_* capability and reverse DNS search in Foreman will be removed.")
-        ip = NicIpResolver.new(:nic => self).to_ip_address(bs) if bs.present?
-        return ip unless ip.nil?
-        failure _("Unable to determine the host's boot server. The DHCP smart proxy failed to provide this information and this subnet is not provided with TFTP services.")
-      rescue => e
-        failure _("failed to detect boot server: %s") % e, e
-      end
+      failure _("Unable to determine boot server. The DHCP smart proxy doesn't provide dhcp_filename capability or failed to provide this information.")
```

Comment:
I would consider moving this to just below `return unless tftp?` so the method becomes:
```ruby
return unless tftp?

unless subnet.dhcp.has_capability?(:DHCP, :dhcp_filename_hostname)
  failure ...
end

tftp.bootServer || URI.parse(subnet.tftp.url).host
```

For simplicity I left out the comments, but those can certainly help.